eventrouter.c

```diff
@@ -1,8 +1,9 @@
 #include "eventrouter/internal/checked_config.h"
 
-#if defined(ER_FREERTOS)
-#include "eventrouter/internal/eventrouter_freertos.c"
-#elif defined(ER_BAREMETAL)
+#ifdef ER_CONFIG_OS
+#include "eventrouter/internal/queue.c"
+#include "eventrouter/internal/eventrouter_os.c"
+#elif ER_IMPLEMENTATION == ER_IMPL_BAREMETAL
```

Comment:
There are many places where `ER_CONFIG_OS` is used to represent code that is common to FreeRTOS and POSIX implementations.